cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataAdmin.java

```diff
@@ -123,6 +124,6 @@ void addProperties(Id.NamespacedId entityId, Map<String, String> properties)
    * @return a {@link Set} records for metadata search.
    * @throws NotFoundException if there is not record found for particular query text.
    */
-  Set<MetadataSearchResultRecord> searchMetadata(String searchQuery, @Nullable MetadataSearchTargetType type)
-    throws NotFoundException;
+  Set<MetadataSearchResultRecord> searchMetadata(String namespaceId, String searchQuery,
```

Comment:
No because Id.NamespacedId does not have the actual namespace id. Need to cast it to right child class.